File: src/week2/P003.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = parseInt(input[0]);

// 두 번째 줄의 값들을 공백으로 나눈 뒤, 각 요소를 숫자로 변환하여 배열에 저장합니다.
const pList = input[1].split(' ').map(Number);

// 1. 배열을 오름차순으로 정렬합니다.
pList.sort((a, b) => a - b);

let result = 0;    // 최종 결과를 저장할 변수
let prefixSum = 0; // 각 단계까지의 누적 합을 저장할 변수

// 2. 누적 합의 총합을 계산합니다.
for (let i = 0; i < n; i++) {
  prefixSum += pList[i]; // 현재까지의 시간을 누적 합에 더합니다.
  result += prefixSum;   // 누적 합을 최종 결과에 더합니다.
}

// 최종 결과를 출력합니다.
console.log(result);

File: src/week2/P003.py
# 11399 ATM 인출 시간 계산하기
n = int(input())
pList = list(map(int, input().split()))

pList.sort()

result = 0
for i in range(1, n+1):
    result += sum(pList[0:i])

print(result)

# 삽입 정렬
n = int(input())
pList = list(map(int, input().split()))

result = 0
for i in range(1, len(pList)):
    for j in range(i, 0, -1):
        if pList[j] < pList[j-1]:
            pList[j], pList[j-1] = pList[j-1], pList[j]
        else:
            break

for i in range(1, n+1):
    result += sum(pList[0:i])
    
print(result)